sentry_sdk/integrations/opentelemetry/sampler.py

```diff
@@ -122,6 +150,9 @@ def sampled_result(span_context, attributes, sample_rate):
     if sample_rate is not None:
         trace_state = trace_state.update(TRACESTATE_SAMPLE_RATE_KEY, str(sample_rate))
 
+    if sample_rand is not None:
+        trace_state = trace_state.update(TRACESTATE_SAMPLE_RAND_KEY, str(sample_rand))
```

Comment:
It is important that we ensure the stringified `sample_rand` has six decimal places here.

Are we already rounding it to six decimal places in the calling code? If yes, please mention in the docstring that the calling code needs to round to six decimal places. If not, then we need to add the rounding logic here, and ideally also add a unit test to verify rounding happens as expected.

In either case, I would use `f"{sample_rand:.6f}"` rather than `str(sample_rand)`.

```suggestion
        sample_rand = sample_rand.quantize(
            Decimal("0.000001"), rounding=ROUND_DOWN, context=Context(prec=6)
        )
        trace_state = trace_state.update(TRACESTATE_SAMPLE_RAND_KEY, f"{sample_rand:.6f}")
```